clustering/infinispan/extension/src/main/java/org/jboss/as/clustering/infinispan/subsystem/InfinispanSubsystemXMLReader.java

```diff
@@ -81,10 +84,13 @@ public void readElement(XMLExtendedStreamReader reader, List<ModelNode> result)
             }
         }
 
+        // Explicitly order operations such that capabilities are defined before they are referenced
+        // This circumvents capability reference integrity issues when operations are not batched
```

Comment:
Kabir can provide some context - as I made this change based on his advise (we did the same for the jgroups changes from last week).  Essentially, this was done solely to make the subsystem test framework happy, since certain tests executes the operations created by the parser one at a time, rather than as a batch - and this will fail unless we order them like this.